Common/SystemTypesResolver.cs

```diff
@@ -15,18 +16,33 @@ public partial class ModuleWeaver
     public void LoadSystemTypes()
     {
         var mscorlib = AssemblyResolver.Resolve(new AssemblyNameReference("mscorlib", null));
-        var typeType = mscorlib?.MainModule.Types.FirstOrDefault(x => x.Name == "Type");
+        var netstandard = AssemblyResolver.Resolve(new AssemblyNameReference("netstandard", null));
+        var runtime = AssemblyResolver.Resolve(new AssemblyNameReference("System.Runtime", null));
+        var core = AssemblyResolver.Resolve(new AssemblyNameReference("System.Core", null));
+
+        var typeType =
```

Comment:
These sets of `FirstOrDefault` are repeated a few times. Please extract these fragments including the null check and `Debug.Write` into a method. Maybe something like `GetTypeFromFirst(params AssemblyReference[] candidateAssemblies)`

Reply:
No problem, will do